rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -5847,34 +5848,151 @@ void validationsForRegisterBtcTransaction_successful() throws IOException, Block
         Assertions.assertTrue(bridgeSupport.validationsForRegisterBtcTransaction(tx.getHash(), height, pmt.bitcoinSerialize(), tx.bitcoinSerialize()));
     }
 
+    private static Stream<Arguments> getTransactionType_sentFromP2SHErpFed_Args() {
+        return Stream.of(
+            // It should be considered a PEGOUT
+            Arguments.of(ActivationConfigsForTest.hop400().forBlock(0), PegTxType.PEGOUT_OR_MIGRATION),
+            // It should be considered a PEGOUT
+            Arguments.of(ActivationConfigsForTest.hop401().forBlock(0), PegTxType.PEGOUT_OR_MIGRATION)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("getTransactionType_sentFromP2SHErpFed_Args")
+    void getTransactionType_sentFromP2SHErpFed(ActivationConfig.ForBlock activations, PegTxType expectedTxType) {
+        // Arrange
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
```

Comment:
Remember that BridgeSupportBuilder uses mocks by default, so no need to declare a new one here if no particular behaviour will be added to it

Reply:
Same thing on the other tests

Reply:
I cannot get the instance of the mock from the builder. I need the instance to return the active federation I want.